### PR TITLE
[resource.images.languageflags.colour] 0.0.3

### PR DESCRIPTION
**Pull request by vdb86**
I tried getting in touch with the original developer (which you can see on his git) but was unsuccessful.
I send a pull request for this fix on Dec 22 2016...

So now I decided to send in this pll request myself, it's only a fix, nothing else.
Serbian flag fixed

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-resources/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by ronie** [comment]
thanx for the contribution, but it isn't correct.

SR = Suriname
RS = Serbia

https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2

**Comment by vdb86** [comment]
Hi Ronie!
Thanks for a quick response.
For some reason opensubtitles shows Serbian subtitles with Suriname flag.
They use this addon to match the language with the flag.
How should I go about fixing this?

**Comment by ronie** [comment]
i'm not sure what the best approach would be, i don't even think there's a 'perfect' solution.

long answer:
opensubtitles uses the ISO 639-3 language representation.
kodi will convert this three letter language tag internally to ISO 639-2 (a 2 letter language tag).

this icon set however uses the ISO_3166-1_alpha-2 standard, which is a 2 letter representation for **countries** (not **languages**!)

and that's where the big problem comes in... there is no reliable way to do a perfect 1:1 match between languages and countries.
for instance, english is spoken in a lot of countries and so is spanish.
in some countries, multiple languages are spoken (switzerland for instance).

the bottom line is, a flag represents a country, not a language.


btw. the opensubtitles addon does not use this resource addon at all.
it's probably the skin you're using that uses this resource addon.

**Comment by vdb86** [comment]
Thank you so much for taking the time to write this response and explain the situation.
I'll probably create a resource that uses ISO 639-2, that way the situation is fixed (or at least it should be and we're generally in a clear).

**Comment by vdb86** [comment]
I see that for example we have:
fr_CA and fr_FR
I presume Kodi will look for French language so we should either have a French flag for fr or have an image that says French, regardless of the geographical location.
Am I on the right path?

**Comment by ronie** [comment]
small update: my response above was incomplete...

i had a closer look at the code of the opensubtitles addon and it returns the language code both as 
ISO 639-1 and as ISO 639-2.

the ISO 639-1 standard is used for the flag icon. so you should be naming your images according to the ISO 639-1 list: https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes

for french, that would be fr.png indeed.

**Comment by vdb86** [comment]
Yeah I figured that one out, but thanks for pointing it out.
Sounds good, I have all the info I need, it'll take me a couple of days but I'll get it done.

**Comment by vdb86** [comment]
Ronie quick question:
What do default Kodi skins use for flags?
I see that they work properly, maybe we should just create a resource out of them?

**Comment by ronie** [comment]
i don't think there's a 'default'...

some skins include their own flag icon set within the skin, while others use a resource addon to provide those flags.

i'd say it's a good idea to create a resource addon with flags for languages ;-)

**Comment by vdb86** [comment]
I can almost swear that confluence used to say ENG, ESP and stuff like that instead of flags.
That would be good to have as well I liked it more...
I already went though the Confluence skin and found just normal flags, the text based ones are not there.
Ideas? :)

**Comment by ronie** [comment]
if a skin wants to display the language as text, you don't need icons for that.
the opensubtitles addons provides the language as a ISO 639-2 text label as well.

i'm not sure if confluence or any other skin uses that though...

**Comment by vdb86** [comment]
Ok, cool, so we only need flags for people who want to use them.
Btw I've found flags in both Confluence and Estuary that ain't correct, so all of this is going to be fixed once and for all! :D

**Comment by vdb86** [comment]
I've created this...
When you get a chance please check it out:
https://github.com/vdb86/resource.images.languageflags.completepack
I'm not 100% about addon.xml...
I'll redo the images in photoshop when I get a chance, but this will work just fine for now.

**Comment by ronie** [comment]
the addon.xml looks ok to me!

please move all the icons directly to the resource folder. (no need for the Flags subfolder).

also, would be great if you could add an info.xml file.
you can simply copy  this one https://github.com/xbmc/repo-resources/blob/jarvis/resource.images.languageflags.colour/info.xml
and put it in the root of your addon.

**Comment by vdb86** [comment]
Sounds good!
How about the icon.png, should that be in the root or the resources folder?

**Comment by ronie** [comment]
i would add it to the root.

**Comment by vdb86** [comment]
All done, I'll submit a pull request in a couple of hours.
Thank you for all your help Ronie!
Now just to figure out how to tell everyone who uses the current resource to change to the new one.
